suzieq/engines/pandas/engineobj.py

```diff
@@ -699,3 +697,29 @@ def _get_user_query_cols(self, query_str: str) -> List[str]:
             return [f for f in table_fields if re.search(rf'\b{f}\b',
                                                          query_str)]
         return []
+
+    def _add_active_to_fields(self, view: str, fields: List[str],
+                              addnl_fields: List[str]) -> None:
+        """Add 'active' field to returned cols
+
+        Depending on the user specification, if we have to return all
+        records within a time window or if view=all, we must also
+        return the active field because it marks if the record was
+        added or deleted
+
+        Args:
+            view(str): value of view specified by user
+            fields(List[str]): the columns to be returned so far
+            addnl_fields(List[str]): the addnl fields to be retrieved so far
+
+        Returns:
+            None
+        """
+
+        if view == "all" or (self.iobj.start_time and self.iobj.end_time):
+            if 'active' not in fields:
+                fields.insert(0, 'active')
+        else:
+            addnl_fields.append('active')
+
+        return
```

Comment:
```suggestion
    def _add_active_to_fields(self, fields: List[str],
                              addnl_fields: List[str]) -> None:
        """Add 'active' field to returned cols
        Depending on the user specification, if we have to return all
        records within a time window or if view=all, we must also
        return the active field because it marks if the record was
        added or deleted
        Args:
            fields(List[str]): the columns to be returned so far
            addnl_fields(List[str]): the addnl fields to be retrieved so far
        Returns:
            None
        """

        if self.iobj.view == "all" or (self.iobj.start_time and self.iobj.end_time):
            if 'active' not in fields:
                fields.insert(0, 'active')
        else:
            addnl_fields.append('active')

        return
```

Reply:
Fixed, also checking that addnl_fields is never None, because we pass that in ospf. Safeguarding it as against None seemed better than passing [] in ospf.py.